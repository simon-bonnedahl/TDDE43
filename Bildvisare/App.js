import { StatusBar } from 'expo-status-bar';
import { Component, useState} from 'react';
import MaterialCommunityIcons from 'react-native-vector-icons/Entypo';
import MaterialCommunityIcons2 from 'react-native-vector-icons/Feather';
import { StyleSheet, View, Image} from 'react-native';


export default function App() {

  var defaultImageSize = 300;
  const [imageId, setImageId] = useState(1);
  const [imageSize, setImageSize] = useState(defaultImageSize);
  const [imageRotation, setImageRotation] = useState(0);

  generateRandomNubmber = function(previous, max){
    while(true){
      let randomNumber = Math.floor(Math.random()*max);
      if(randomNumber != previous){
        return randomNumber;
      }
    }
  }

  return (
    <View style={styles.container}>
       <Image  
          style={{transform: [{ rotate: imageRotation + 'deg'}]}} 
          testID='image'
          source= {{
            testID:'image-style',
            rotate: imageRotation,
            width: imageSize,
            height: imageSize,
            uri: "https://picsum.photos/id/" + imageId + "/" + defaultImageSize}}/>
        <Image />
        <View style={styles.icons}>
        <MaterialCommunityIcons
            name="cycle"
            testID='refresh-button'
            size={40}
            color={'#000000'} 
            onPress={() => setImageId(generateRandomNubmber(imageId, 300))}/>
        <MaterialCommunityIcons
            name="plus" 
            testID='plus-button'
            color={'#000000'} 
            size={40}
            onPress={() => setImageSize(imageSize+10)}/>
           
            <MaterialCommunityIcons
            name="minus" 
            testID='minus-button'
            color={'#000000'} 
            size={40}
            onPress={() => setImageSize(imageSize-10)}/>
           
          
            <MaterialCommunityIcons2
            name="rotate-ccw" 
            testID='rotate-left-button'
            color={'#000000'} 
            size={40}
            onPress={() => setImageRotation(imageRotation-90)}/>

            <MaterialCommunityIcons2
            name="rotate-cw"
            testID='rotate-right-button' 
            color={'#000000'} 
            size={40}
            onPress={() => setImageRotation(imageRotation+90)}/>
          
           
           <StatusBar/>
        </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#999'
  },icons:{
    flexDirection: 'row'
  }
 
});
