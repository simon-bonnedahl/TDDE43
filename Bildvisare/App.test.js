import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen, fireEvent} from '@testing-library/react-native'

import App from './App';

describe('<App />', () => {
    
    it('renders correctly', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    
    it('refresh button works', async () => {
        render(<App/>)
        const image = await screen.findByTestId('image')
        const uriBefore = image.props['source'].uri
        await fireEvent.press(screen.getByTestId('refresh-button'))
        const uriAfter = image.props['source'].uri
        expect(uriAfter).toEqual(expect.not.stringMatching(uriBefore))
        
    })
    it('plus button works', async ()=>  {
        render(<App/>)
        const image = await screen.findByTestId('image')
        const sizeBefore = image.props['source'].width
        await fireEvent.press(screen.getByTestId('plus-button'))
        const sizeAfter = image.props['source'].width
        expect(sizeAfter).toBe(sizeBefore+10)

    })
    it('minus button works', async ()=>{
        render(<App/>)
        const image = await screen.findByTestId('image')
        const sizeBefore = image.props['source'].width
        await fireEvent.press(screen.getByTestId('minus-button'))
        const sizeAfter = image.props['source'].width
        expect(sizeAfter).toBe(sizeBefore-10)
    })
    it('rotate-left button works', async ()=>{
        render(<App/>)
        const image = await screen.findByTestId('image')
        const rotationBefore = image.props['source'].rotate
        await fireEvent.press(screen.getByTestId('rotate-left-button'))
        const rotationAfter = image.props['source'].rotate
        expect(rotationAfter).toBe(rotationBefore-90)
    })
    it('rotate-right button works', async ()=>{
        render(<App/>)
        const image = await screen.findByTestId('image')
        const rotationBefore = image.props['source'].rotate
        await fireEvent.press(screen.getByTestId('rotate-right-button'))
        const rotationAfter = image.props['source'].rotate
        expect(rotationAfter).toBe(rotationBefore+90)
    })

    
});

