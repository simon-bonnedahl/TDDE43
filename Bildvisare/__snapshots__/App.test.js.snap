// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#999",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "height": 300,
        "rotate": 0,
        "testID": "image-style",
        "uri": "https://picsum.photos/id/1/300",
        "width": 300,
      }
    }
    style={
      Object {
        "transform": Array [
          Object {
            "rotate": "0deg",
          },
        ],
      }
    }
    testID="image"
  />
  <Image />
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text />
    <Text />
    <Text />
    <Text />
    <Text />
  </View>
</View>
`;
